[Bug]: [nullable vector]Streaming node panic: unsupported data type FloatVector when nullable FloatVector field with BM25 function

## Environment

- **Milvus Version**: master-20260324-a18d125b-amd64
- **Deployment Mode**: cluster (with streaming enabled, woodpecker service mode)
- **MQ**: woodpecker (streaming.enabled=true, streaming.woodpecker.embedded=false)
- **SDK**: pymilvus v2.7.0rc155
- **OS**: Linux (Kubernetes)

## Reproduction

### Option A: Script

```python
import random
from pymilvus import MilvusClient, DataType, Function, FunctionType

client = MilvusClient(uri="http://localhost:19530")

collection_name = "test_nullable_vec_bug"
client.drop_collection(collection_name)

schema = client.create_schema()
schema.add_field("id", DataType.INT64, is_primary=True, auto_id=False)
schema.add_field("text", DataType.VARCHAR, max_length=65535,
                 enable_analyzer=True, enable_match=True)
# nullable FloatVector
schema.add_field("vector", DataType.FLOAT_VECTOR, dim=5, nullable=True)
# BM25 output
schema.add_field("sparse", DataType.SPARSE_FLOAT_VECTOR)

bm25 = Function(
    name="bm25_fn",
    function_type=FunctionType.BM25,
    input_field_names=["text"],
    output_field_names=["sparse"],
)
schema.add_function(bm25)

index_params = client.prepare_index_params()
index_params.add_index(field_name="vector", index_type="FLAT", metric_type="IP")
index_params.add_index(field_name="sparse", index_type="SPARSE_INVERTED_INDEX",
                       metric_type="BM25")

client.create_collection(
    collection_name=collection_name,
    schema=schema,
    index_params=index_params,
)
print("Collection created")

# Don't pass "vector" field at all -> nullable field becomes null
data = []
for i in range(2000):
    data.append({
        "id": i,
        "text": " ".join([f"word{random.randint(1, 10000)}" for _ in range(500)]),
    })

print(f"Inserting {len(data)} rows without vector field (null)...")
res = client.insert(collection_name=collection_name, data=data)
print(res)
```

### Trigger Conditions

- **Frequency**: always
- **First observed after**: immediately after insert
- **Does NOT happen when**: FloatVector field is not nullable, or BM25 function is not present

All three conditions must be met to trigger:
1. Collection has a BM25 function (embeddingNode exists in pipeline)
2. FloatVector field is set to `nullable=True`
3. Insert data with FloatVector field as null (not passing the field)

In streaming mode, multiple messages within the same segment trigger `MergeFieldData`, which hits the unsupported type path.

## Expected Behavior

Insert should succeed without panic. Nullable FloatVector fields with null values should be handled gracefully by the embedding node and streaming pipeline.

## Actual Behavior

Streaming node panics with:

```
panic: unsupported data type: FloatVector
```

The panic causes a crash loop — Kubernetes restarts the streaming node, which reads the same WAL data from woodpecker segments, hits the same panic, and enters a `CrashLoopBackOff` cycle. This cascading failure also destabilizes query nodes (they fail to connect to etcd due to reconnection storms).

## Error Logs

```
panic: unsupported data type: FloatVector

panic({0x8105a00?, 0xc001647848?})
	/usr/local/go/src/runtime/panic.go:811 +0x168
```

Streaming node logs also show woodpecker read failures as a consequence:

```
[WARN] [segment/segment_handle.go:551] ["read batch failed on all quorum nodes"]
  [logName=by-dev-rootcoord-dml_15] [error="no more data"]

[WARN] [segment/segment_handle.go:518] ["read batch failed on node"]
  [node=woodpecker-pod-kill-6-milvus-woodpecker-0...] [error="no more data"]
```

## Analysis Hints

- Suspect code location: `embedding_node.go:262` and `pipeline.go:93`, `stream_pipeline.go:83` based on the panic stack trace
- The panic occurs in `MergeFieldData` when the embedding node encounters a FloatVector data type it doesn't handle
- The issue is specific to the streaming pipeline path — the embeddingNode needs to either skip or properly handle nullable vector fields with null values
- Crash-loop is particularly severe because the WAL (woodpecker) replays the same data on restart, causing repeated panics

--- a/pkg/util/typeutil/schema.go
+++ b/pkg/util/typeutil/schema.go
@@ -2095,6 +2095,8 @@ func MergeFieldData(dst []*schemapb.FieldData, src []*schemapb.FieldData) error 
 				} else {
 					dstVector.GetVectorArray().Data = append(dstVector.GetVectorArray().Data, srcVector.VectorArray.Data...)
 				}
+			case nil:
+				// nullable vector field where all rows are null — no vector data to merge
 			default:
 				return errors.New("unsupported data type: " + srcFieldData.Type.String())
 			}

--- a/pkg/util/typeutil/schema_test.go
+++ b/pkg/util/typeutil/schema_test.go
@@ -2124,6 +2124,46 @@ func TestMergeFieldData(t *testing.T) {
 		err := MergeFieldData([]*schemapb.FieldData{emptyField}, []*schemapb.FieldData{emptyField})
 		assert.Error(t, err)
 	})
+
+	t.Run("nullable float vector - all null src", func(t *testing.T) {
+		dim := int64(4)
+		dstFields := []*schemapb.FieldData{
+			{
+				Type:      schemapb.DataType_FloatVector,
+				FieldName: "nullable_vec",
+				FieldId:   200,
+				ValidData: []bool{true},
+				Field: &schemapb.FieldData_Vectors{
+					Vectors: &schemapb.VectorField{
+						Dim: dim,
+						Data: &schemapb.VectorField_FloatVector{
+							FloatVector: &schemapb.FloatArray{
+								Data: []float32{1, 2, 3, 4},
+							},
+						},
+					},
+				},
+			},
+		}
+		// src: all rows are null -> VectorField.Data = nil
+		srcFields := []*schemapb.FieldData{
+			{
+				Type:      schemapb.DataType_FloatVector,
+				FieldName: "nullable_vec",
+				FieldId:   200,
+				ValidData: []bool{false},
+				Field: &schemapb.FieldData_Vectors{
+					Vectors: &schemapb.VectorField{
+						Dim:  dim,
+						Data: nil, // all null -> no FloatVector oneof set
+					},
+				},
+			},
+		}
+
+		err := MergeFieldData(dstFields, srcFields)
+		assert.NoError(t, err, "MergeFieldData should handle nullable FloatVector with nil VectorField.Data")
+	})
 }
 
 type FieldDataSuite struct {

